redisinsight/ui/src/pages/workbench/components/wb-results/WBResults/WBResults.tsx

```diff
@@ -101,7 +90,16 @@ const WBResults = (props: Props) => {
           resultsMode={resultsMode}
           db={db}
           onQueryOpen={() => onQueryOpen(id)}
-          onQueryProfile={handleQueryProfile}
+          onQueryProfile={(profileType: ProfileQueryType) => {
+            const profileQuery = generateProfileQueryForCommand(command, profileType)
+            if (profileQuery) {
+              return onQueryProfile(
+                profileQuery,
+                null,
+                { mode, results: resultsMode, clearEditor: false, },
+              )
+            }
+          }}
```

Comment:
@romansergeenkosofteq Moving back to closure due to the function's entirety depending on the closure variables.

Reply:
You have 3 not declatered variables:
![image](https://user-images.githubusercontent.com/82098578/218755182-c62c1403-194e-4bb9-aed9-c4e487c00088.png)
Just pass this to your function
```
const handleQueryProfile = (
    profileType: ProfileQueryType,
    commandExecution: { command: string, mode: RunQueryMode, resultsMode: ResultsMode }
  ) => {
    const { command, mode, resultsMode } = commandExecution
    const profileQuery = generateProfileQueryForCommand(command, profileType)
    if (profileQuery) {
      onQueryProfile(
        profileQuery,
        null,
        { mode, results: resultsMode, clearEditor: false, },
      )
    }
  }
```